js/dom.js

```diff
@@ -1,57 +1,82 @@
+/* eslint-disable max-classes-per-file */
 const container = document.querySelector('#books-list');
 
-function createArticle(books, index) {
+class Book {
+  constructor(id, title, author) {
+    this.id = id;
+    this.title = title;
+    this.author = author;
+  }
+}
+
+class Books {
+  constructor() {
+    this.list = [];
+  }
+
+  addBook(book) {
+    this.list.push(book);
+  }
+
+  removeBook(book) {
+    this.list.splice(this.list.indexOf(book), 1);
+  }
+
+  copyList(list) {
+    this.list = list;
+  }
+}
+
+const books = new Books();
+
+function displayArticle(book) {
   const article = document.createElement('article');
-  article.setAttribute('id', `article-${index}`);
+  article.setAttribute('id', `article-${book.id}`);
   const title = document.createElement('p');
-  title.textContent = books[index].title;
+  title.textContent = book.title;
   article.appendChild(title);
   const author = document.createElement('p');
-  author.textContent = books[index].author;
+  author.textContent = book.author;
   article.appendChild(author);
-  const button = document.createElement('button');
-  button.classList.add('remove-btn');
-  button.textContent = 'Remove';
-  button.addEventListener('click', () => {
+  const remButton = document.createElement('button');
+  remButton.classList.add('remove-btn');
+  remButton.textContent = 'Remove';
+  remButton.addEventListener('click', () => {
     article.remove();
-    books.splice(index, 1);
-    localStorage.setItem('books', JSON.stringify(books));
+    books.removeBook(book);
+    localStorage.setItem('books', JSON.stringify(books.list));
   });
-  article.appendChild(button);
+  article.appendChild(remButton);
   const hr = document.createElement('hr');
   article.appendChild(hr);
   container.appendChild(article);
 }
 
-function addBook(books, title, author) {
-  if (title !== '' && author !== '') {
-    const book = {
-      title,
-      author,
-    };
-    books.push(book);
-    createArticle(books, books.length - 1);
-    localStorage.setItem('books', JSON.stringify(books));
+function displayBooksList(books) {
+  for (let i = 0; i < books.list.length; i += 1) {
+    displayArticle(books.list[i]);
   }
 }
 
-function displayBooksList(books) {
-  for (let i = 0; i < books.length; i += 1) {
-    createArticle(books, i);
+function createArticle(books, title, author) {
+  if (title !== '' && author !== '') {
+    const book = new Book(books.length, title, author);
+    books.addBook(book);
+    displayArticle(book);
+    localStorage.setItem('books', JSON.stringify(book.list));
```

Comment:
* I see undefined in the local storage when i add books kindly look into this 🙏 
![image](https://user-images.githubusercontent.com/52497246/133547162-807e4b94-ff2c-40c6-9429-c128d1f26467.png)